tests/qat/test_quantum_backend.py

```diff
@@ -714,6 +714,163 @@ def test_create_duration_timeline_mapping(self):
             *[PostProcessing] * 4,
         ]
 
+    @pytest.mark.parametrize(
+        "hw", [get_test_model(), get_default_echo_hardware(), get_default_RTCS_hardware()]
+    )
+    def test_duration_timeline_times(self, hw):
+        # Tests that the creation of a duration timeline with a two-qubit circuit
+        # gives a timeline where each pulse is concurrent.
+        engine = hw.create_engine()
+        q1 = hw.get_qubit(0)
+        q2 = hw.get_qubit(1)
+
+        # Build a simple circuit
+        builder = (
+            get_builder(hw)
+            .X(q1, np.pi / 2.0)
+            .X(q2, np.pi)
+            .cnot(q1, q2)
+            .measure_mean_z(q1)
+            .measure_mean_z(q2)
+        )
+        res1 = engine.create_duration_timeline(builder.instructions)
+
+        # Check the times match up
+        for _, val in res1.items():
+            end = 0
+            for pos in val:
+                assert end == pos.start
+                end = pos.end
```

Comment:
this for loop only tests chaining of start, end, start, end on a single target.
Could you add more checks that contrasts a timeline of a target against another ?

Reply:
Sorry, my comment for the test made it confusing. What I meant was: the test checks that instructions don't overlap or leave gaps in time (so the end of one instruction should be the beginning of the next).

But I agree, having a test that that compares targets where instructions should align will be valuable. I'll add that :smile: